mapper/orm/src/main/java/org/hibernate/search/mapper/orm/loading/impl/AbstractHibernateOrmLoadingStrategy.java

```diff
@@ -54,17 +51,37 @@ public HibernateOrmQueryLoader<E, I> createQueryLoader(List<LoadingTypeContext<?
 				includedTypesFilter.add( typeContext.typeIdentifier().javaClass() );
 			}
 		}
+		EntityMappingType commonSuperType = toMostSpecificCommonEntitySuperType( typeContexts );
```

Comment:
We probably need to test for null here, and throw an exception? We have that in the subclass...